<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>OpenInfraredViewer</class>
 <widget class="QMainWindow" name="OpenInfraredViewer">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1213</width>
    <height>835</height>
   </rect>
  </property>
  <property name="baseSize">
   <size>
    <width>1280</width>
    <height>720</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Open Infrared Viewer</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_3">
    <item row="0" column="1">
     <widget class="QLabel" name="canvasLabel">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>80</width>
        <height>60</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::Panel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Sunken</enum>
      </property>
      <property name="text">
       <string>Viewport Canvas</string>
      </property>
      <property name="scaledContents">
       <bool>false</bool>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab1">
       <attribute name="title">
        <string>General</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="1">
         <widget class="QWidget" name="exportWidget" native="true">
          <layout class="QGridLayout" name="gridLayout_7">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item row="0" column="1">
            <widget class="QPushButton" name="recordButton">
             <property name="text">
              <string>Start Recording</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QPushButton" name="captureButton">
             <property name="text">
              <string>Capture</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="9" column="1">
         <widget class="QGroupBox" name="transformGroupBox">
          <property name="title">
           <string>Transform</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_13">
           <item row="0" column="0">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Rotate</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QComboBox" name="rotateComboBox">
             <item>
              <property name="text">
               <string>0ยบ</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>90ยบ</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>180ยบ</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>270ยบ</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>Flip</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QComboBox" name="flipComboBox">
             <item>
              <property name="text">
               <string>None</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Vertically</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Horizontally</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Both Axis</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QGroupBox" name="groupBox_2">
          <property name="title">
           <string>Flat Field Correction</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_11">
           <item row="1" column="0">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>FFC Trigger Mode</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QComboBox" name="ffcModeComboBox"/>
           </item>
           <item row="0" column="0" colspan="2">
            <widget class="QPushButton" name="triggerFfcButton">
             <property name="text">
              <string>Trigger FFC</string>
             </property>
            </widget>
           </item>
           <item row="4" column="0" colspan="2">
            <widget class="QCheckBox" name="freezeViewportCheckBox">
             <property name="toolTip">
              <string>Avoids viewport flashing when the shutter gets in the way of the sensor.</string>
             </property>
             <property name="text">
              <string>Freeze Viewport on FFC</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QGroupBox" name="manualSpanGroupBox">
          <property name="title">
           <string>Manual Span</string>
          </property>
          <property name="flat">
           <bool>false</bool>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="checked">
           <bool>false</bool>
          </property>
          <layout class="QGridLayout" name="gridLayout_9">
           <item row="7" column="0">
            <widget class="QDoubleSpinBox" name="spanEndDoubleSpinBox">
             <property name="decimals">
              <number>1</number>
             </property>
             <property name="minimum">
              <double>-65536.000000000000000</double>
             </property>
             <property name="maximum">
              <double>65536.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>100.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="5" column="1">
            <widget class="QSlider" name="spanStartSlider">
             <property name="value">
              <number>9</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="invertedAppearance">
              <bool>false</bool>
             </property>
             <property name="invertedControls">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_11">
             <property name="text">
              <string>Slider Maximum</string>
             </property>
            </widget>
           </item>
           <item row="5" column="0">
            <widget class="QDoubleSpinBox" name="spanStartDoubleSpinBox">
             <property name="decimals">
              <number>1</number>
             </property>
             <property name="minimum">
              <double>-65536.000000000000000</double>
             </property>
             <property name="maximum">
              <double>65536.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>100.000000000000000</double>
             </property>
             <property name="value">
              <double>0.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="6" column="0" colspan="2">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Span Range End</string>
             </property>
            </widget>
           </item>
           <item row="4" column="0" colspan="2">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Span Range Start</string>
             </property>
            </widget>
           </item>
           <item row="7" column="1">
            <widget class="QSlider" name="spanEndSlider">
             <property name="value">
              <number>90</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="invertedAppearance">
              <bool>false</bool>
             </property>
             <property name="invertedControls">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label_10">
             <property name="text">
              <string>Slider Minimum</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QDoubleSpinBox" name="sliderRangeMinimumDoubleSpinBox">
             <property name="decimals">
              <number>1</number>
             </property>
             <property name="minimum">
              <double>-65536.000000000000000</double>
             </property>
             <property name="maximum">
              <double>65536.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>100.000000000000000</double>
             </property>
             <property name="value">
              <double>0.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QDoubleSpinBox" name="sliderRangeMaximumDoubleSpinBox">
             <property name="decimals">
              <number>1</number>
             </property>
             <property name="minimum">
              <double>-65536.000000000000000</double>
             </property>
             <property name="maximum">
              <double>65536.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>100.000000000000000</double>
             </property>
             <property name="value">
              <double>0.000000000000000</double>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="8" column="1">
         <widget class="QGroupBox" name="groupBox_3">
          <property name="title">
           <string>Color Palette</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_12">
           <item row="1" column="0">
            <widget class="QCheckBox" name="invertPaletteCheckBox">
             <property name="text">
              <string>Invert</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QComboBox" name="colorPaletteComboBox"/>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="paletteRulerLabel">
             <property name="minimumSize">
              <size>
               <width>256</width>
               <height>20</height>
              </size>
             </property>
             <property name="frameShape">
              <enum>QFrame::NoFrame</enum>
             </property>
             <property name="text">
              <string>Palette Ruler</string>
             </property>
             <property name="scaledContents">
              <bool>true</bool>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="10" column="1">
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab2">
       <attribute name="title">
        <string>Calibration</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="1" column="0">
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="0" column="0">
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>Blind Pixel Detection</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_4">
           <item row="0" column="0">
            <widget class="QPushButton" name="blindPixelDetectionButton">
             <property name="text">
              <string>Start Blind Pixel Detection</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QCheckBox" name="checkBox">
             <property name="text">
              <string>Highlight Blind Pixels</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1213</width>
     <height>32</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>Calibration</string>
    </property>
    <addaction name="actionCorrect_Bad_Pixels"/>
    <addaction name="actionNon_Uniformity_Correction"/>
    <addaction name="actionRadiometry_Calibration"/>
    <addaction name="actionFFC_Calibration"/>
   </widget>
   <widget class="QMenu" name="menuComposition">
    <property name="title">
     <string>Composition</string>
    </property>
    <addaction name="actionColor_Pallete"/>
    <addaction name="actionShow_Color_Scale"/>
    <addaction name="actionShow_Bad_Pixels"/>
   </widget>
   <widget class="QMenu" name="menuSettings">
    <property name="title">
     <string>Settings</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuFile_2">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen_Device"/>
   </widget>
   <addaction name="menuFile_2"/>
   <addaction name="menuFile"/>
   <addaction name="menuComposition"/>
   <addaction name="menuSettings"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionCorrect_Bad_Pixels">
   <property name="text">
    <string>Bad Pixel Correction</string>
   </property>
  </action>
  <action name="actionNon_Uniformity_Correction">
   <property name="text">
    <string>Non Uniformity Correction</string>
   </property>
  </action>
  <action name="actionRadiometry_Calibration">
   <property name="text">
    <string>Radiometry Calibration</string>
   </property>
  </action>
  <action name="actionFFC_Calibration">
   <property name="text">
    <string>Flat Field Correction</string>
   </property>
  </action>
  <action name="actionColor_Pallete">
   <property name="text">
    <string>Color Pallete</string>
   </property>
  </action>
  <action name="actionShow_Color_Scale">
   <property name="text">
    <string>Show Color Scale</string>
   </property>
  </action>
  <action name="actionShow_Bad_Pixels">
   <property name="text">
    <string>Show Bad Pixels</string>
   </property>
  </action>
  <action name="actionOpen_Device">
   <property name="text">
    <string>Open Device</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
 <slots>
  <slot>record()</slot>
  <slot>pause()</slot>
  <slot>stop()</slot>
  <slot>enablePreview(bool)</slot>
  <slot>configureCaptureSettings()</slot>
  <slot>takeImage()</slot>
  <slot>startCamera()</slot>
  <slot>toggleLock()</slot>
  <slot>setMuted(bool)</slot>
  <slot>stopCamera()</slot>
  <slot>setExposureCompensation(int)</slot>
 </slots>
</ui>
